### PR TITLE
add ManagedIdentitySettings to session pool 2024-10-02-preview

### PR DESCRIPTION
**Pull request by najian**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [ ] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [ ] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by ms-bogdan-rotaru** [comment]
Please fix the errors reported by Swagger PrettierCheck.

**Review by rkmanda — DISMISSED** [review]
:shipit:

**Comment by rkmanda** [comment]
Pl fix the errors here - seems like there is a parsing error for the open api spec json
https://github.com/Azure/azure-rest-api-specs/pull/30771/checks?check_run_id=30846335360

For the patch... rule that is breaking please add a suppression..

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/SessionPools.json [hunk 1]
```diff
@@ -634,16 +642,56 @@
...
+      "properties": {
+        "identity": {
+          "description": "The resource ID of a user-assigned managed identity that is assigned to the Session Pool, or 'system' for system-assigned identity.",
+          "type": "string"
+        },
+        "lifecycle": {
+          "description": "Use to select the lifecycle stages of a Session Pool during which the Managed Identity should be available.",
+          "enum": [
+            "None",
+            "Main"
```
Comment (bowen5) [inline specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/SessionPools.json:676]: In current Session scenario, suppose `Main` indicates the whole lifecycle, right? Am thinking whether we should use `All` or `Main` here, considering the different iteration story of [`Main`, `None`] -> [`Main`, `None`, `All`] vs [`All`, `None`] -> [`All`, `None`, `Main`]. Would like you hear more of your thoughts

Comment (najian) [reply specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/SessionPools.json:676]: From PM "for sessions, if we have pullImage and main, then if someone selects main, it's not clear if that includes image pull as well 
i think none and main sound good for sessions"

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/examples/SessionPools_CreateOrUpdate.json [hunk 2]
```diff
@@ -38,11 +38,21 @@
...
+          "registryCredentials": {
+            "server": "test.azurecr.io",
+            "identity": "/subscriptions/7a497526-bb8d-4816-9795-db1418a1f977/resourcegroups/test/providers/Microsoft.ManagedIdentity/userAssignedIdentities/testSP"
           }
         },
         "sessionNetworkConfiguration": {
           "status": "EgressEnabled"
-        }
+        },
+        "managedIdentitySettings": [
```
Comment (jijohn14) [inline specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/examples/SessionPools_CreateOrUpdate.json:50]: These are missing in GET example

**Review by jijohn14 — CHANGES_REQUESTED** [review]

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/examples/SessionPools_CreateOrUpdate.json (hunk 2 above)
Comment (najian) [reply specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/examples/SessionPools_CreateOrUpdate.json:53]: Added

File: package-lock.json [hunk 3]
```diff
@@ -95,14 +95,15 @@
       }
     },
     "eng/tools/node_modules/vite": {
```
Comment (jijohn14) [inline package-lock.json:97]: What this file changed?

File: specification/app/resource-manager/readme.md [hunk 4]
```diff
@@ -66,6 +66,11 @@ input-file:
   - Microsoft.App/preview/2024-10-02-preview/SourceControls.json
   - Microsoft.App/preview/2024-10-02-preview/Subscriptions.json
   - Microsoft.App/preview/2024-10-02-preview/Usages.json
+directive:
+  - suppress: PatchBodyParametersSchema
+    from: SessionPools.json
+    reason: |
+      Session Pool is using componentType as the discriminator. While the discriminator is a required property, this rule prevent it being present in the patch request body.
```
Comment (jijohn14) [inline specification/app/resource-manager/readme.md:73]: Why does session pool not support patch?

**Review by jijohn14 — CHANGES_REQUESTED** [review]

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/SessionPools.json [hunk 5]
```diff
@@ -634,16 +642,56 @@
...
+      "type": "object",
+      "required": [
+        "identity"
+      ],
+      "properties": {
+        "identity": {
+          "description": "The resource ID of a user-assigned managed identity that is assigned to the Session Pool, or 'system' for system-assigned identity.",
+          "type": "string"
+        },
+        "lifecycle": {
```
Comment (Juliehzl) [inline specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/SessionPools.json:672]: should it be lifeCycle to align with enum value?

File: specification/app/resource-manager/readme.md (hunk 4 above)
Comment (najian) [reply specification/app/resource-manager/readme.md:73]: It do support patch, but during patch(common type) identity is not required and in common type it marked as required. This is suppressed suggested by ARM reviewer above [rkmanda](https://github.com/rkmanda) "For the patch... rule that is breaking please add a suppression.."

Comment (jijohn14) [reply specification/app/resource-manager/readme.md:73]: Thank you for the explanation

**Review by jijohn14 — APPROVED** [review]

**Pull request merged**